src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient_private.cs

```diff
@@ -106,11 +109,40 @@ Uri BuildUriForLocksRoute(string key, string label)
             return builder.Uri;
         }
 
-        void BuildBatchQuery(UriBuilder builder, SettingSelector selector)
+        private string EscapeReservedCharacters(string input)
+        {
+            string resp = string.Empty;
+            for (int i=0; i<input.Length; i++)
+            {
+                if (ReservedCharacters.Contains(input[i]))
+                {
+                    resp += $"\\{input[i]}";
+                }
+                else
+                {
+                    resp += input[i];
+                }
+            }
+            return resp;
+        }
+
+        internal void BuildBatchQuery(UriBuilder builder, SettingSelector selector)
         {
             if (selector.Keys.Count > 0)
             {
-                var keys = string.Join(",", selector.Keys).ToLower();
+                var keysCopy = new List<string>();
+                foreach (var key in selector.Keys)
+                {
+                    if (key.IndexOfAny(ReservedCharacters) != -1)
+                    {
+                        keysCopy.Add(EscapeReservedCharacters(key));
```

Comment:
Would it be better to always call EscapeReservedCharacters? IndexOfAny already needs to parse the string, making it always call would be cleaner to read.

Reply:
I'll change it